test/e2e/tsconfig.json

```diff
@@ -1,7 +1,9 @@
 {
   "compilerOptions": {
-    "target": "es2016" /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */,
-    "module": "commonjs" /* Specify what module code is generated. */,
+    "target": "es2022" /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */,
+    "lib": ["es2023"],
```

Comment:
Are these compatible with recent versions of vscode?

Reply:
This is based on:

* VSCode is on Node 20
* The tsconfig for working with Node 20 suggests the target and lib selection above: https://www.npmjs.com/package/@tsconfig/node20

However, this was only tentative, it should all be reviewed once the all the parts allow a run through of Slang with WASM.

There is a question about whether we should target Node 18 rather than Node 20 so that we are keeping to the Hardhat node version support promise. This wouldn't affect the VSCode library but you can run the language server independently.

Reply:
I think that if it works with vscode is enough. We don't need to keep the same support, as vscode updates on a different cadence.